src/Kafka/Settings/Settings.tsx

```diff
@@ -0,0 +1,157 @@
+        </Card>
+      </PageSection>
+      <Page>
+        <PageSection>
+          <Card>
+            <CardBody>
+              <Flex>
+                <FlexItem component={"span"} spacer={{ default: "spacerSm" }}>
+                  <strong>
+                    {t("settings.connection_re_authentication_title")}
+                  </strong>
+                </FlexItem>
+                <FlexItem>
+                  {(() => {
+                    switch (connectionStatus) {
+                      case "On":
+                        return t(
+                          "settings.Connection_re_authentication_states.turnon"
+                        );
+                      case "TurningOff":
+                        return (
+                          <>
+                            <Spinner
+                              size="md"
+                              aria-valuetext={t(
+                                "settings.Connection_re_authentication_states.turning_off"
+                              )}
+                              aria-label={t(
+                                "settings.connection_re_authentication_label"
+                              )}
+                            />{" "}
+                            {t(
+                              "settings.Connection_re_authentication_states.turning_off"
+                            )}
+                          </>
+                        );
+                      case "Off":
+                        return t(
+                          "settings.Connection_re_authentication_states.turnoff"
+                        );
+                      case "TurningOn":
+                        return (
+                          <>
+                            <Spinner
+                              size="md"
+                              aria-valuetext={
+                                "settings.Connection_re_authentication_states.turning_on"
+                              }
+                              aria-label={t(
+                                "settings.connection_re_authentication_label"
+                              )}
+                            />{" "}
+                            {t(
+                              "settings.Connection_re_authentication_states.turning_on"
+                            )}
+                          </>
+                        );
+                      default:
+                        return null;
+                    }
+                  })()}
+                </FlexItem>
+                <FlexItem component={"span"} align={{ default: "alignRight" }}>
+                  <Switch
+                    id="Connection-re-authentication-switch"
+                    aria-label={t(
+                      "settings.connection_re_authentication_switch"
+                    )}
+                    isChecked={
+                      connectionStatus === "On" ||
+                      connectionStatus === "TurningOff"
```

Comment:
Does this mean that if the status is "TurningOff" or "TurningOn" that I can click the switch again? Should we set a disabled prop if connection status is "TurningOff" or "TurningOn" while we wait for the API response?